app/controllers/foreman_inventory_upload/uploads_controller.rb

```diff
@@ -14,14 +14,7 @@ def download_file
       files = Dir["{#{ForemanInventoryUpload.uploads_file_path(filename)},#{ForemanInventoryUpload.done_file_path(filename)}}"]
 
       return send_file files.first, disposition: 'attachment', filename: filename unless files.empty?
-      throw_flash_error "File doesn't exist"
-    end
-
-    def throw_flash_error(message)
-      process_error(
-        :redirect => foreman_inventory_upload_index_path,
-        :error_msg => message
-      )
+      raise ActionController::BadRequest.new("The report file doesn't exist")
```

Comment:
Better to use either `::Foreman::Exception` or `::Foreman::WrappedException`. I think they are shown in a better way, plus we have a way to mark them in documentation.

Reply:
updated, thanks @ShimShtein !